master/internal/task/allocation.go

```diff
@@ -1159,6 +1162,16 @@ func (a *Allocation) getModelState() model.AllocationState {
 	return *a.model.State
 }
 
+func (a *Allocation) ready() bool {
+	// Most trials use `a.rendezvous` and the normal rendezvous APIs, and go through this path.
+	return (a.rendezvous != nil && a.rendezvous.ready()) ||
```

Comment:
So when is rendezvous used?   I do not see it triggered with 2 node dtrain job (cifar10_pytorch).

Reply:
it is used in the non-hpc mode. it is https://github.com/determined-ai/determined/blob/master/harness/determined/exec/prep_container.py#L69 vs https://github.com/determined-ai/determined/blob/master/harness/determined/exec/prep_container.py#L82